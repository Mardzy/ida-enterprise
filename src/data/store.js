import { fromJS } from "immutable";
import { createEpicMiddleware } from "redux-observable";
import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import persistState from "redux-localstorage";
import { routerMiddleware } from "connected-react-router/immutable";

import { INITIAL_STATE } from "./initial-state";
import { rootReducer, history } from "./reducer";
import { rootEpic } from "./epic";
// import { gaMiddleware } from "data/config/analytics";
import { serialize, deserialize, merge, slice } from "./config/persistent-store";

/**
 * Construct an epic middleware.
 * @type {EpicMiddleware<Action, any, any, Action>}
 */
const epicMiddleware = createEpicMiddleware();

/**
 * Middleware to apply in the store.
 * @type {*[]}
 */
const middleware = [
  // gaMiddleware,
  epicMiddleware,
  routerMiddleware(history)
];

/**
 * Configuration for keeping all the redux state in localstorage.
 * @type {{key: string, slicer: slice, serialize: serialize, deserialize: deserialize, merge: merge}}
 */
const persistStateConfig = {
  key: "partner-dashboard",
  slicer: slice,
  serialize, deserialize, merge
};

/**
 * Create a Redux store.
 * @param {Object} initialState
 * @return {Store<any>}
 */
const configureStore = (initialState = {}) => createStore(
  rootReducer,
  INITIAL_STATE.merge(fromJS(initialState)),
  composeWithDevTools({})(
    applyMiddleware(...middleware),
    persistState(["auth", "farms"], persistStateConfig)
  )
);

/**
 * Create a store with the reducers.
 */
const store = configureStore();

epicMiddleware.run(rootEpic);

export default store;
export { history, epicMiddleware, configureStore };