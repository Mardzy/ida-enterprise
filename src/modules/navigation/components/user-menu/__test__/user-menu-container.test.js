import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import store from "../../../../../data/store";
import UserMenuContainer from "../user-menu-container";
import authTypes from "../../../../../data/auth/types";

const mockStore = configureStore([]);

describe("The user menu container", () => {

  it("displays the user's full name after it has been loaded", () => {
    const testStore = mockStore({});

    store.dispatch({
      type: authTypes.REQUEST_USER.FULFILLED,
      payload: {
        username: "sandor@connecterra.io",
        givenName: "Sandor",
        surName: "Kocsis"
      }
    });

    const wrapper = mount(
      <Provider store={store}>
        <UserMenuContainer/>
      </Provider>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

});