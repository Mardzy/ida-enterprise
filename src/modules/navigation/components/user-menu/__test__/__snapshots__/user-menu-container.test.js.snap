// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The user menu container displays the user's full name after it has been loaded 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(lifecycle(UserMenu))>
    <lifecycle(UserMenu)
      fetchUser={[Function]}
      isLoading={false}
      onLogoutClicked={[Function]}
      user={
        Immutable.Map {
          "username": "sandor@connecterra.io",
          "givenName": "Sandor",
          "surName": "Kocsis",
        }
      }
    >
      <UserMenu
        fetchUser={[Function]}
        isLoading={false}
        onLogoutClicked={[Function]}
        user={
          Immutable.Map {
            "username": "sandor@connecterra.io",
            "givenName": "Sandor",
            "surName": "Kocsis",
          }
        }
      >
        <MenuItem>
          <div
            className="item"
            onClick={[Function]}
          >
            <MenuHeader>
              <div
                className="header"
              >
                Sandor Kocsis
                 
              </div>
            </MenuHeader>
            <MenuMenu>
              <div
                className="menu"
              >
                <MenuItem
                  onClick={[Function]}
                >
                  <a
                    className="item"
                    onClick={[Function]}
                  >
                    Log out
                  </a>
                </MenuItem>
              </div>
            </MenuMenu>
          </div>
        </MenuItem>
      </UserMenu>
    </lifecycle(UserMenu)>
  </Connect(lifecycle(UserMenu))>
</Provider>
`;
